src/worker/start.rs

```diff
@@ -57,6 +61,21 @@ pub struct WorkerStartOpts {
     manager: ManagerOpts,
 }
 
+fn launcher_setup(task: &Task, def: LauncherDefinition) -> tako::Result<ProgramDefinition> {
+    let allocation = task.resource_allocation().unwrap();
```

Comment:
`?`

Reply:
launcher setup should be executed only on a running task and such that it has an allocation. It is a fatal internal error if allocation is not there, so I decided for unwrap.

Reply:
Ok. I suggest to always use `expect` in such cases so that we have documentation about why it should not fail.